packages/hadron-document/src/element.ts

```diff
@@ -719,6 +720,48 @@ export class Element extends EventEmitter {
     this._bubbleUp(Events.Reverted, this);
   }
 
+  /**
+   * Expands only the target element
+   */
+  expand(): void {
+    this.expanded = true;
+    this._bubbleUp(ElementEvents.Expanded, this);
+  }
+
+  /**
+   * Collapses only the target element
+   */
+  collapse(): void {
+    this.expanded = false;
+    this._bubbleUp(ElementEvents.Collapsed, this);
+  }
+
+  /**
+   * Expands the target element and all of its sub-elements
+   */
+  expandAllFields(): void {
+    this.expanded = true;
+    if (this.elements) {
+      for (const element of this.elements) {
+        element.expandAllFields();
+      }
+    }
+    this.emit(ElementEvents.Expanded, this);
+  }
+
+  /**
+   * Collapses the target element and all of its sub-elements
+   */
+  collapseAllFields(): void {
+    this.expanded = false;
+    if (this.elements) {
+      for (const element of this.elements) {
+        element.collapseAllFields();
+      }
+    }
+    this.emit(ElementEvents.Collapsed, this);
+  }
```

Comment:
So while I see why you have expandAll method separate from expand, collapse separately from collapseAll doesn't make much sense to me. In what case we'd ever want parent to collapse, but children stay expanded?